File: backend/app/schemas/purchase.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

PurchaseSize = Literal["pequeña", "mediana", "grande"]


class PurchaseItemIn(BaseModel):
    product_id: str
    size: PurchaseSize = "mediana"
    quantity: int = Field(gt=0)
    unit_cost: int = Field(gt=0)          # centavos
    new_price: int | None = Field(None, gt=0)   # centavos; si falta se calcula por markup


class PurchaseCreate(BaseModel):
    supplier: str | None = None
    reference: str | None = None
    note: str | None = None
    items: list[PurchaseItemIn] = Field(min_length=1)


class PurchaseItemOut(BaseModel):
    product_id: str
    title: str
    size: PurchaseSize
    quantity: int
    unit_cost: int
    prev_cost: int | None = None
    prev_price: int | None = None
    new_price: int | None = None
    markup_pct: float | None = None


class PurchaseSummary(BaseModel):
    purchase_id: str
    tenant_id: str
    supplier: str | None = None
    reference: str | None = None
    item_count: int
    total_units: int
    total_cost: int
    created_at: datetime


class PurchaseDetail(PurchaseSummary):
    note: str | None = None
    items: list[PurchaseItemOut]
    updated_at: datetime | None = None

File: frontend/src/types/purchase.ts
export type PurchaseSize = "pequeña" | "mediana" | "grande";

export interface PurchaseItem {
  product_id: string;
  title: string;
  size: PurchaseSize;
  quantity: number;
  unit_cost: number;
  prev_cost: number | null;
  prev_price: number | null;
  new_price: number | null;
  markup_pct: number | null;
}

export interface PurchaseSummary {
  purchase_id: string;
  tenant_id: string;
  supplier: string | null;
  reference: string | null;
  item_count: number;
  total_units: number;
  total_cost: number;
  created_at: string;
}

export interface PurchaseDetail extends PurchaseSummary {
  note: string | null;
  items: PurchaseItem[];
  updated_at: string | null;
}

export interface PurchaseItemInput {
  product_id: string;
  size: PurchaseSize;
  quantity: number;
  unit_cost: number;
  new_price?: number | null;
}

export interface PurchaseCreateInput {
  supplier?: string | null;
  reference?: string | null;
  note?: string | null;
  items: PurchaseItemInput[];
}
